Rx/v2/src/rxcpp/operators/rx-merge_delay_error.hpp

```diff
@@ -0,0 +1,302 @@
+
+                                // when the out observer is unsubscribed all the
+                                // inner subscriptions are unsubscribed as well
+                                auto innercstoken = state->out.add(innercs);
+
+                                innercs.add(make_subscription([state, innercstoken](){
+                                        state->out.remove(innercstoken);
+                                }));
+
+                                auto selectedSource = state->coordinator.in(st);
+
+                                ++state->pendingCompletions;
+                                // this subscribe does not share the source subscription
+                                // so that when it is unsubscribed the source will continue
+                                auto sinkInner = make_subscriber<value_type>(
+                                        state->out,
+                                        innercs,
+                                // on_next
+                                        [state, st](value_type ct) {
+                                                state->out.on_next(std::move(ct));
+                                        },
+                                // on_error
+                                        [state](std::exception_ptr e) {
+                                                if(--state->pendingCompletions == 0) {
+                                                        state->out.on_error(e);
+                                                } else {
+                                                        state->exception.add(e);
+                                                }
+                                        },
+                                //on_completed
+                                        [state](){
+                                                if (--state->pendingCompletions == 0) {
+                                                        if(!state->exception.empty()) {
+                                                            state->out.on_error(
+                                                                std::make_exception_ptr(std::move(state->exception)));
+                                                        } else {
+                                                                state->out.on_completed();
+                                                        }
+                                                }
+                                        }
+                                );
+
+                                auto selectedSinkInner = state->coordinator.out(sinkInner);
+                                selectedSource.subscribe(std::move(selectedSinkInner));
+                        },
+                // on_error
+                        [state](std::exception_ptr e) {
+			                if(--state->pendingCompletions == 0) {
+			                    state->out.on_error(e);
+			                } else {
```

Comment:
perhaps when the last stream terminates in an error the error should be added to the `composite_exception` and then the `composite_exception` is sent to `out.on_error`?